src/contrib/cluster/Akka.Cluster.Sharding/ShardSupervisionStrategy.cs

```diff
@@ -0,0 +1,68 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="ShardSupervisionStrategy.cs" company="Akka.NET Project">
+//      Copyright (C) 2009-2022 Lightbend Inc. <http://www.lightbend.com>
+//      Copyright (C) 2013-2025 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Akka.Actor;
+using Akka.Actor.Internal;
+
+namespace Akka.Cluster.Sharding;
+
+public class ShardSupervisionStrategy: OneForOneStrategy
+{
+    public ShardSupervisionStrategy(int? maxNrOfRetries, TimeSpan? withinTimeRange, Func<Exception, Directive> localOnlyDecider) 
+        : base(maxNrOfRetries, withinTimeRange, localOnlyDecider)
+    {
+    }
+
+    public ShardSupervisionStrategy(int? maxNrOfRetries, TimeSpan? withinTimeRange, IDecider decider) 
+        : base(maxNrOfRetries, withinTimeRange, decider)
+    {
+    }
+
+    public ShardSupervisionStrategy(int maxNrOfRetries, int withinTimeMilliseconds, Func<Exception, Directive> localOnlyDecider, bool loggingEnabled = true) 
+        : base(maxNrOfRetries, withinTimeMilliseconds, localOnlyDecider, loggingEnabled)
+    {
+    }
+
+    public ShardSupervisionStrategy(int maxNrOfRetries, int withinTimeMilliseconds, IDecider decider, bool loggingEnabled = true) 
+        : base(maxNrOfRetries, withinTimeMilliseconds, decider, loggingEnabled)
+    {
+    }
+
+    public ShardSupervisionStrategy(Func<Exception, Directive> localOnlyDecider) 
+        : base(localOnlyDecider)
+    {
+    }
+
+    public ShardSupervisionStrategy(Func<Exception, Directive> localOnlyDecider, bool loggingEnabled = true) 
+        : base(localOnlyDecider, loggingEnabled)
+    {
+    }
+
+    public ShardSupervisionStrategy(IDecider decider) 
+        : base(decider)
+    {
+    }
+
+    public ShardSupervisionStrategy()
+    {
+    }
+
+    public override void ProcessFailure(IActorContext context, bool restart, IActorRef child, Exception cause, ChildRestartStats stats, IReadOnlyCollection<ChildRestartStats> children)
+    {
+        if (restart && stats.RequestRestartPermission(MaxNumberOfRetries, WithinTimeRangeMilliseconds))
+            RestartChild(child, cause, suspendFirst: false);
+        else
+        {
+            // Only send failure feedback if restart was requested
+            if(restart)
+                context.Self.Tell(new ExcessiveSupervisorRestartPassivation(child, WithinTimeRangeMilliseconds, MaxNumberOfRetries, cause));
```

Comment:
The `ProcessFailure` code is pretty much the same as the `OneToOneStrategy` with these additional lines, we send the `Shard` actor a warning message that this failing child is due for termination because it is thrashing.

Reply:
This is not quite right - we also need to handle scenarios where the `SupervisorStrategy` decided to issue a `Stop` directive. Reason being: if the actor failed in such a way that it has to be stopped, it's by definition an irrecoverable exception. Continuing to remember the entity after we get this type of signal back is net-destructive.

Reply:
Also: we need to make sure this only applies to entity actors, not to any other children of the `Shard`, such as the RE infrastructure itself. You can determine this by checking the actor paths.

Reply:
> Also: we need to make sure this only applies to entity actors, not to any other children of the Shard, such as the RE infrastructure itself. You can determine this by checking the actor paths.

nevermind, this gets handled inside the `Shard` message handlers by the looks of things